In [2]:
from urllib.request import urlopen
from urllib.parse import quote_plus
from bs4 import BeautifulSoup

from selenium import webdriver
import time
import pandas as pd
import requests

In [6]:
driver = webdriver.Chrome()
url = 'https://www.youtube.com/watch?v=nTTsOgUXAfQ&ab_channel=%EC%BC%80%EC%9D%B8TV'
driver.get(url)

last_page_height = driver.execute_script('return document.documentElement.scrollHeight')

while True:
    driver.execute_script('window.scrollTo(0, document.documentElement.scrollHeight);')
    time.sleep(3.0)
    new_page_height = driver.execute_script('return document.documentElement.scrollHeight')
    
    if new_page_height == last_page_height:
        break
    last_page_height = new_page_height
    

html_source = driver.page_source

driver.close()

soup = BeautifulSoup(html_source, 'lxml')

youtube_user_ID = soup.select('div#header-author > h3 > a > span')
youtube_comment = soup.select('yt-formatted-string#content-text')



str_youtube_user_ID = []
str_youtube_comment = []

for i in range(len(youtube_user_ID)):
    str_temp = str(youtube_user_ID[i].text)    
    str_temp = str_temp.replace('\n', '')
    str_temp = str_temp.replace('\t', '')
    str_temp = str_temp.replace('                ','')
    str_youtube_user_ID.append(str_temp)
    
    str_temp = str(youtube_comment[i].text)
    str_temp = str_temp.replace('\n', '')
    str_temp = str_temp.replace('\t', '')
    str_temp = str_temp.replace('                ','')
    str_youtube_comment.append(str_temp)
    

    


In [7]:
for i in range(len(str_youtube_user_ID)):
    print(str_youtube_user_ID[i], str_youtube_comment[i])

              수리 Suri             환갑을 바라보는 연세에도 도전하시는 모습이 진짜 뭉클하네요... 응원합니다!
              이름없음             점점 숫자가 터무니 없어지는게 킬포임 ㅋㅋㅋㅋㅋㅋ
              _식빵             케조씨 저렇게 빵터진거 처음봤다 맨이야이렇게 잘웃으면 얼마나좋아 코도 더 커보이고
              BESKIA             지맘대로 합성해서 지원시켜버리는거 줫나웃기네 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ
              qualification             01:37웅장한 배경과 베짱있는 자세, 화면 등장까지 완벽하다 ㅋㅋ진짜 잘만드셨네 ㅋㅋㅌㅌㅋ
              Wildcat             기록 수치가 존나 근본없는게 개웃김ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ
              사용자             가짜 사나이 2기 0화보고 왔습니다.케인님이 참여 못하신 것이 아쉽다고 생각합니다.하긴 사이클 2일 3시간하고 달리기 마하 9로 가시는 케인님이 참여하셨다면조교들이 오히려 체력이 딸려 욕먹을까봐 케인님이 불참하신게 아닌가 싶어 이해하고 있습니다.
              ㅂㅅㅂ             이렇게 코가 터지게 웃으시는 건 첨 봐요이건 귀하네요...
              김정민             진짜 케인님 웃으시니까 넘 보기좋다ㅋㅋ 영상만드시는분도 맨날 타지리 김성근보다 저런 영상위주로 만들어주시는게 재미있는것같아요 엿날팹시맨처럼요ㅋㅋㅋ
              PS PS             케인인님 빵터져서 웃는모습 보기좋다맨이야~ 마지막극찬 까지 완벽 다시한번만보자래... 만든사람 뿌듯하겠다
              강현석             이 영상을 계기로 케인인님에 대해서 알게 되었는데 진짜ㅋㅋㅋㅋㅋㅋ이렇게 활짝 웃는 모습 또 보고싶어요
              청국장             집

In [8]:
df = pd.DataFrame(data = str_youtube_user_ID, columns = ['ID'])
df2 = pd.DataFrame(data = str_youtube_comment, columns = ['Comment'])

fin = pd.concat([df, df2], axis = 1)
fin

ID  \
0                         수리 Suri               
1                            이름없음               
2                             _식빵               
3                          BESKIA               
4                   qualification               
...                                       ...   
1205                   Sam Smiith               
1206                           쇼어               
1207                     tina Lon               
1208                          양명규               
1209                      Du Stok               

                                                Comment  
0            환갑을 바라보는 연세에도 도전하시는 모습이 진짜 뭉클하네요... 응원합니다!  
1                           점점 숫자가 터무니 없어지는게 킬포임 ㅋㅋㅋㅋㅋㅋ  
2         케조씨 저렇게 빵터진거 처음봤다 맨이야이렇게 잘웃으면 얼마나좋아 코도 더 커보이고  
3           지맘대로 합성해서 지원시켜버리는거 줫나웃기네 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ  
4     01:37웅장한 배경과 베짱있는 자세, 화면 등장까지 완벽하다 ㅋㅋ진짜 잘만드셨네 ...  
...                                                 ...  
1205                                                 Sz  
1206                            제발 내 유튜브에 보이지마 꼴보기 싫어 ㅠ  
1207                 본인이 하면 재미 남이 하면 고소한국 사회 참 정겨워졌네 ㅋㅋ  
1208                                              모지리같냐  
1209                                               Fake  

[1210 rows x 2 columns]

In [12]:
fin.to_csv('케인댓글수집.csv', encoding = 'utf-8')

- 유튜브 영상 댓글은 1800여개 수집된 댓글은 1210개 >> 약 600여개는 ***댓글의 답글***임
- 수집해보자

In [ ]:
url = 'https://www.youtube.com/watch?v=nTTsOgUXAfQ&ab_channel=%EC%BC%80%EC%9D%B8TV'
source = requests.get(url)

soup = BeautifulSoup(source)